Mathlib/AlgebraicGeometry/OpenImmersion.lean

```diff
@@ -186,6 +186,24 @@ instance basic_open_isOpenImmersion {R : CommRingCat.{u}} (f : R) :
     exact Spec_map_localization_isIso R (Submonoid.powers f) x
 #align algebraic_geometry.Scheme.basic_open_IsOpenImmersion AlgebraicGeometry.Scheme.basic_open_isOpenImmersion
 
+instance {R} [CommRing R] (f : R) :
+    IsOpenImmersion (Spec.map (CommRingCat.ofHom (algebraMap R (Localization.Away f)))) :=
+  basic_open_isOpenImmersion (R := .of R) f
+
+lemma IsOpenImmersion.of_isLocalization {R S} [CommRing R] [CommRing S]
+    [Algebra R S] (f : R) [IsLocalization.Away f S] :
+    IsOpenImmersion (Spec.map (CommRingCat.ofHom (algebraMap R S))) := by
```

Comment:
Haha, I was about to ask whether we have this :) A challenge (which I don't need) is to prove the same for localization at a f.g. submonoid.

Reply:
I don't think it is that hard actually. The same proof shows that the stalks are isomorphic and the underlying map is a topological embedding. To show that the image is open, we just need to show that it is the intersection of the basic open of the generators, which can't be that hard in lean right...?

Reply:
Yeah probably not too hard. I was thinking about using induction and transitivity of open immersion and of localization.